File: arrays/partitionarray.py
class Solution:
    def partitionArray(self, nums: List[int], k: int) -> int:
        nums.sort()
        
        count = 0
        currentMin = nums[0]

        for i in range(len(nums)):
            if nums[i] - currentMin > k:
                currentMin = nums[i]
                count += 1
        
        return count + 1
        

File: arrays/partitionarray.cpp
class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        vector<int> ans(nums.size(),pivot);
        int a=0;
        for(int i=0;i<nums.size();i++){
            if(nums[i]>pivot) a++;}
        int x=0,y=nums.size()-a;
        for(int i:nums){
            if(i<pivot){
                ans[x]=i;
                x++;}
            else if(i>pivot){
                ans[y]=i;
                y++;}}
        return ans;
    }
};
